sdk/python/kfp/components/pipeline_task.py

```diff
@@ -456,6 +470,56 @@ def my_pipeline():
             self._task_spec.dependent_tasks.append(task.name)
         return self
 
+    def set_exit_task(self, exit_task: 'PipelineTask') -> 'PipelineTask':
+        """Specifies the task to invoke after the current task exits,
+        irrespective of whether the current task completes successfully.
+
+        Args:
+            exit_task: Task to execute after this task is done
+
+        Example:
+          ::
+
+            @dsl.pipeline()
+            def my_pipeline(sample_input1: str = 'message'):
+                task = fail_op(message=sample_input1)
+                clean_up = print_op(message=task.output)
+                task.set_exit_task(exit_task=clean_up)
+        """
+        from kfp.components import pipeline_context
+        from kfp.components import tasks_group
+
+        for _, input_spec in exit_task.component_spec.inputs.items():
+            if not input_spec.optional:
+                raise ValueError(
+                    f'Clean up task {exit_task.name} requires a default value to make sure the Exit handler never fails.'
+                )
+
+        for task in pipeline_context.Pipeline.get_default_pipeline(
+        ).tasks.values():
+            if isinstance(
+                    task,
+                    tasks_group.ExitHandler) and task.exit_task == exit_task:
+                raise ValueError(
+                    f'Clean up task can not be used for multiple worker tasks. Clean up task {exit_task.name} already used for {task.name}'
```

Comment:
I think worker tasks gets the idea across, but it's a new term. To keep language consistent in the user-facing error messages, what do you think about using `upstream task` and `exit task` here? There is a precedent for using the terms upstream/downstream to delineate between two tasks: https://github.com/kubeflow/pipelines/blob/66569960b53c56c5c341deb341a29cf351ec7578/sdk/python/kfp/compiler/compiler_utils.py#L659-L663